File: backend/src/main/java/com/mini/credit/dto/epargne/OperationEpargneResponse.java
package com.mini.credit.dto.epargne;

import com.mini.credit.enums.ModePaiement;
import com.mini.credit.enums.SensOperation;
import com.mini.credit.enums.TypeOperationEpargne;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Builder
public class OperationEpargneResponse {
    private Long id;
    private Long compteEpargneId;
    private String numeroCompte;
    private Long membreId;
    private String membreNomComplet;
    private Long demandeCreditId;
    private Long creditId;
    private LocalDateTime dateOperation;
    private TypeOperationEpargne typeOperation;
    private BigDecimal montant;
    private SensOperation sens;
    private ModePaiement modePaiement;
    private String referenceExterne;
    private Long agentId;
    private Long sessionCaisseId;
    private String observation;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/app/features/epargne/models/operation-epargne-response.ts
import { ModePaiement } from './mode-paiement';

export type TypeOperationEpargne =
  | 'COTISATION'
  | 'EPARGNE'
  | 'RETRAIT'
  | 'BLOCAGE_GARANTIE'
  | 'DEBLOCAGE_GARANTIE'
  | 'AJUSTEMENT';

export type SensOperation = 'ENTREE' | 'SORTIE';

export interface OperationEpargneRequest {
  compteEpargneId: number;
  membreId: number;
  dateOperation: string;
  typeOperation: TypeOperationEpargne;
  montant: number;
  sens?: SensOperation;
  modePaiement?: ModePaiement;
  referenceExterne?: string;
  agentId?: number;
  sessionCaisseId?: number;
  createdBy?: number;
  observation?: string;
}

export interface OperationEpargneResponse {
  id: number;
  compteEpargneId: number;
  numeroCompte: string;
  membreId: number;
  membreNomComplet: string;
  dateOperation: string;
  typeOperation: TypeOperationEpargne;
  montant: number;
  sens: SensOperation;
  modePaiement?: ModePaiement;
  referenceExterne?: string;
  agentId?: number;
  sessionCaisseId?: number;
  observation?: string;
  createdAt: string;
  updatedAt: string;
}